src/DotNet.Status.Web/tests/DotNet.Status.Web.Tests/TelemetryControllerTests.cs

```diff
@@ -0,0 +1,75 @@
+using Xunit;
+using DotNet.Status.Web.Controllers;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Moq;
+using Microsoft.AspNetCore.Mvc;
+using Kusto.Ingest;
+using System.Threading.Tasks;
+
+namespace DotNet.Status.Web.Tests
+{
+    public class TelemetryControllerTests
+    {
+        [Fact]
+        public void TestArcadeValidationTelemetryCollection()
+        {
+            var mockLoggerMock = new Mock<ILogger<TelemetryController>>();
+            var telemetryOptionsMock = new Mock<IOptionsSnapshot<TelemetryOptions>>();
+            var telemetryOptions = new TelemetryOptions
+            {
+                KustoIngestConnectionString = "fakekustoconnectionstring",
+                KustoDatabase = "fakekustodatbase"
+            };
+            telemetryOptionsMock.SetupGet(x => x.Value).Returns(telemetryOptions);
+            var kustoIngestResultMock = new Mock<IKustoIngestionResult>();
+            var kustoIngestClientMock = new Mock<IKustoIngestClient>();
+            kustoIngestClientMock.Setup(x => x.IngestFromStreamAsync(It.IsAny<System.IO.Stream>(), It.IsAny<KustoIngestionProperties>(), null))
+                .Returns(Task.FromResult(kustoIngestResultMock.Object));
+            var controller = new TelemetryController(mockLoggerMock.Object, telemetryOptionsMock.Object, kustoIngestClientMock.Object);
```

Comment:
I'd use DI here, since it will handle a lot of the logicistics of IOptions and ILoggers for you (and also means you can tweak that stuff without having to mess with every single test)

```suggestion
            var collection = new ServiceCollection();
            collection.AddOptions();
            collection.AddLogging(l => {
                // This makes the logs appear in the XUnit output, super handy for debugging failures
                l.AddProvider(new XUnitProvider(_output)
            });
            collection.Configure<TelemetryOptions>(o => {
                 o.KustoIngestConnectionString = "fakekustoconnectionstring",
                 o.KustoDatabase = "fakekustodatbase"
            });
            collection.AddScoped<TelemetryController>();
            await using ServiceProvider services = collection.BuildServiceProvider();
            var controller = collection.GetRequiredService<TelemetryController>();
```

Reply:
Shoved all this into a method to handle DI and mocks.